### ISSUE TITLE
Archive KidneySingleCellAtlas data in ENA

### ISSUE THREAD

**Issue by mshadbolt** [ebi-ait/hca-ebi-wrangler-central#18]
This dataset has metadata archived at BioStudies and BioSamples but is yet to be archived in ENA

The latest spreadsheet is here:

https://drive.google.com/open?id=1vk5e2yPk6plMnoor1JvcLxzxwUbTmTrs

It has a related AUDR ticket here:

https://github.com/HumanCellAtlas/hca-data-wrangling/issues/341

But AUDR issues should have been fixed in the latest spreadsheet before the r-release.

The data is in DCP1 DSS

I am not sure if fastqs have already been converted to bams (@ESapenaVentura ?)

See this issue for previous work https://github.com/HumanCellAtlas/hca-data-wrangling/issues/427

**Comment by ESapenaVentura** [comment]
Refer to this https://github.com/HumanCellAtlas/hca-data-wrangling/issues/427 for the current status

tl;dr: Only SequencingRuns are left to upload.

**Comment by ESapenaVentura** [comment]
**Update**: Just ran the command 
```
docker run -v $PWD:/app/output --env INGEST_API_URL=https://api.ingest.archive.data.humancellatlas.org/ --env DSP_API_URL=https://submission.ebi.ac.uk --env AAP_API_URL=https://api.aai.ebi.ac.uk/auth --env ONTOLOGY_API_URL=https://www.ebi.ac.uk/ols --env AAP_API_DOMAIN=subs.team-2 --env AAP_API_USER=<aap_user> --env AAP_API_PASSWORD=<aap_pass> --env VALIDATION_POLL_FOREVER=False --env SUBMISSION_POLL_FOREVER=False quay.io/ebi-ait/ingest-archiver:d2020-07-09.1_CLI --alias_prefix=HCA --ingest_submission_uuid=d5410c6e-612d-421a-a66f-2de5e04dd050 --exclude_types=project,study,samples,sequencingExperiment
```

**Comment by ESapenaVentura** [comment]
**Update**: Ran the ingest-file-archiver, files are being converted and uploaded.

**Comment by ESapenaVentura** [comment]
Submission with the 22 sequencingRuns: [DSP submission](https://submission.ebi.ac.uk/api/submissions/b794dcdc-2df5-4f3a-962e-d6edc6ffa651)

**Comment by ESapenaVentura** [comment]
**Update**: 1 file left for upload!

**Comment by ESapenaVentura** [comment]
Done https://www.ebi.ac.uk/ena/browser/view/PRJEB37165

AI before closing the ticket:
- [ ] Make sure ingest retrieved the accessions successfully
- [ ] Make sure the accessions are put back in the metadata spreadsheet

**Comment by ESapenaVentura** [comment]
https://ui.ingest.archive.data.humancellatlas.org/submissions/detail?uuid=d5410c6e-612d-421a-a66f-2de5e04dd050&project=abe1a013-af7a-45ed-8c26-f3793c24a1f4

Seems like there is a problem with the accessions. Only the submittables from the latest DSP submission are recorded (20 SequencingRuns), and no accession has been retrieved for them. @aaclan-ebi ?

**Comment by aaclan-ebi** [comment]
Sorry, I wasn't able to add pagination in the UI. It only shows the first 20 archive entities.

The sequencing runs weren't updated because the DSP submission processing result states values are 'Error' instead of 'Complete'. I've adjusted the archiver script to still update the archive entities and metadata in Ingest. The files were now updated with sequencing run accessions.

https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/5daf405071fe4a0008e325ae/files

The other accessions (project, study, sequencing experiment) can be added by doing an update submission.  The archiver now uses the archive entities to track which metadata to update. Since those metadata were submitted to dsp before the archive entities records were added in Ingest, the archive entities were not created. We can do migration and add those archive entities by inspecting the dsp submission however that is not implemented yet. The easiest and fastest way is to do an update submission.

**Comment by ESapenaVentura** [comment]
Thanks for looking into this @aaclan-ebi ! So, just to clarify, next steps are:

1. Create an update submission uploading the latest spreadsheet (With accessions) to ingest.
1. Hit submit on the archiving option

Please confirm so I can begin doing this! thanks

**Comment by aaclan-ebi** [comment]
Please just do no. 1 for now. Don't hit the submit yet. 

Can I get back to you on this? I think we need to do special case handling for DCP submissions which were archived and updated with accessions.

**Comment by aaclan-ebi** [comment]
I am going to file issues to investigate the following:

1. How to export DCP datasets which have updates (new accessions) ? The submissions are already in Complete state.

2. For DCP and new datasets which were archived prior the archiver support for storing accessions in Ingest. Those have accessions but the accessions may not have been put back to Ingest yet. How should we put back accessions in Ingest?

**Comment by aaclan-ebi** [comment]
I've created this issue to be able to put back the accessions in Ingest entities.

ebi-ait/hca-ebi-dev-team#226

**Comment by ESapenaVentura** [comment]
There are already issues filled to investigate the steps left so I think this ticket has filled its purpose.

I'm closing the ticket but if you do not agree feel free to open.

**Comment by lauraclarke** [comment]
With the new board, this is being reopened to reflect the fact the EMBL-EBI accessions from this archiving process need to be added back to the ingest record of this project

This ticket can be closed when said archive accessions are added back to the record so it can then be propagated to the DCP

**Comment by ESapenaVentura** [comment]
Update release - VERY LOW PRIORITY, only run accessions are missing and it's a DCP1 dataset (Low value/High cost)

**Comment by idazucchi** [comment]
This is the same dataset as #81

Both experiment and run accessions are already present - I think this can be considered done
